include/swift/Sema/IDETypeChecking.h

```diff
@@ -206,6 +206,36 @@ namespace swift {
   /// the decl context.
   ProtocolDecl *resolveProtocolName(DeclContext *dc, StringRef Name);
 
+  /// Reported type of a variable declaration.
+  struct VariableTypeInfo {
+    /// The start of the variable identifier.
+    uint32_t Offset;
+
+    /// The length of the variable identifier.
+    uint32_t Length;
+
+    /// Whether the variable has an explicit type annotation.
+    bool HasExplicitType;
+
+    /// The start of the printed type in a separate string buffer.
+    uint32_t TypeOffset;
+
+    /// The length of the printed type in a separate string buffer.
+    uint32_t TypeLength;
```

Comment:
If we avoid the conversion and use `VariableTypeInfo` in `LangSupport.h`, wouldn't that introduce a dependency on `IDETypeChecking`?

Reply:
True. We could move the `VariableTypeInfo` type further into a module that’s imported by both `Sema` and `SourceKit` (e.g. `AST`) but that seems like an overkill to me. Still, I think it would be nice if both types had the same layout.